src/libraries/Microsoft.Extensions.Http/src/DefaultHttpMessageHandlerBuilder.cs

```diff
@@ -28,7 +28,26 @@ public DefaultHttpMessageHandlerBuilder(IServiceProvider services)
             }
         }
 
-        public override HttpMessageHandler PrimaryHandler { get; set; } = new HttpClientHandler();
+        private HttpMessageHandler? _primaryHandler;
+        internal bool PrimaryHandlerIsSet { get; private set;}
```

Comment:
Why is this needed? Why can't `_primaryHandler` just be checked directly to see whether it's null?

Reply:
Because there are existing cases when primary handler is set explicitly to null. E.g. [in gRPC](https://github.com/grpc/grpc-dotnet/blob/854e0fd3d0a7da8507ae0f4bf3adac129fb7e10b/src/Grpc.Net.ClientFactory/GrpcClientServiceExtensions.cs#L349-L354). While I kinda consider this pattern erroneous, I still don't want to break the users...